Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1389,8 +1389,38 @@ void GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrack(GridTrack& track
             RenderGrid* inner = downcast<RenderGrid>(gridItem);
             GridSpan span = m_renderGrid->gridSpanForChild(*gridItem, m_direction);
 
+            auto extraMarginFromSubgridAncestorGutters = [&]() -> std::optional<LayoutUnit> {
+                if (span.startLine() != trackIndex && span.endLine() - 1 != trackIndex)
+                    return std::nullopt;
+
+                auto gutterTotal = 0_lu;
+                const auto* currentAncestor = dynamicDowncast<RenderGrid>(gridItem->parent());
+                auto flowAwareDirection = iterator.direction();
+
+                while (currentAncestor && currentAncestor->isSubgrid(flowAwareDirection)) {
```

Comment:
I wouldn't block on this, but it seems that a lot of the complexity here is iterating up the sub grid ancestors and converting coordinate space.

It would be somewhat nice if we had a `forEachSubgridAncestor` iterator thing that we could use in all the places we traverse the sub grid ancestors, which put all of that code in one spot.

Reply:
Yeah I agree. We probably do this in a few other places and should definitely have something. I'll try to see if I can come up with such a thing and try to insert it where it is needed in the current code